src/layout/node.ts

```diff
@@ -74,36 +82,58 @@ export default function(node: Node, source: Source): void {
             if (element.namespaceURI === Constant.SVG_NAMESPACE) { tag = Constant.SVG_PREFIX + tag; }
 
             switch (tag) {
+                case "HTML":
+                    // Skip processing nodes within iframe if they were already removed from the DOM before we got to processing them.
+                    // In those cases, we skip them and continue processing rest of the nodes.
+                    let parentFrame = insideFrame && parent ? dom.iframe(parent) : null;
+                    if (insideFrame && parentFrame === null) { break; }
+                    let htmlData = { tag, attributes: getAttributes(element.attributes) };
+                    dom[call](node, parentFrame ? parentFrame : parent, htmlData, source);
+                    break;
                 case "SCRIPT":
                 case "NOSCRIPT":
                 case "META":
                     break;
                 case "HEAD":
                     let head = { tag, attributes: getAttributes(element.attributes) };
                     // Capture base href as part of discovering DOM
-                    head.attributes[Constant.BASE_TAG] = location.protocol + "//" + location.hostname;
+                    if (insideFrame === false) { head.attributes[Constant.BASE_ATTRIBUTE] = location.protocol + "//" + location.hostname; }
```

Comment:
>) [](start = 45, length = 1)

Why not capture this in an iframe if it exists?

Reply:
We will capture it if it exists but we won't force generate the base tag. By default iframe will accquire parent's base tag so that works. However if IFRAME had it's own base, we would grab it like any other DOM element.